### ISSUE TITLE
`dotnet_sort_system_directives_first` does not generate error even if `IDE0055` severity is set to `error`

### ISSUE THREAD

**Issue by bastbu** [dotnet/roslyn#57644]
**Version Used**:

.NET SDK 6.0.100

**Steps to Reproduce**:

1. Create a new project with a `.editorconfig` equivalent to:

```
[*.cs]
dotnet_diagnostic.IDE0055.severity = error
dotnet_sort_system_directives_first = true
```

2. Set `<EnforceCodeStyleInBuild>true<EnforceCodeStyleInBuild/>` in the `.csproj` file

3. Add the following code to the project:

```csharp
using Newtonsoft.Json;
using System;

public static class Foo
{
    public static void Bar()
    {
        var result = JsonConvert.DeserializeObject<int>("1");
        Console.WriteLine( result);
    }
}
``` 

**Expected Behavior**:

An error should be generated for the additional space at `Console.WriteLine( result);`. Another error should be generated because the `using`s are not sorted as defined in the `.editorconfig`.

The [IDE0055 documentation](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/formatting-rules#organize-using-directives) implies that organizing of using directives is part of `IDE0055`, and hence I would expect that violations to `dotnet_sort_system_directives_first` are treated as violations to `IDE0055`.

**Actual Behavior**:

The only error generated is for the missing space in `Console.WriteLine( result);`.

**Comment by sharwell** [comment]
IDE0055 does not consider `dotnet_sort_system_directives_first` (or any other sorting of using directives). Marking as a question to identify the diagnostic (if any) which applies for incorrectly sorted using directives.

**Comment by dartignan** [comment]
Hello,
Is it possible to clarify the final diagnostic/answer to this issue marked as completed?
As of today, the problem is still present.

**Comment by CyrusNajmabadi** [comment]
hi @dartignan , as stated above IDE0055 (formatting) is unrelated to 'sorting'.  Formatting only affects whitespace/newlines between tokens.  It doesn't restructure the order of things in the code.

**Comment by dartignan** [comment]
Thanks @CyrusNajmabadi for the clarification. The formatting definition you describe totally makes sense.
What is misleading is that `dotnet_sort_system_directives_first` is classified under `IDE0055` in [Microsoft Documentation](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/dotnet-formatting-options).

**Comment by Santas** [comment]
How can this rule trigger an error?
